request_a_govuk_domain/request/admin.py

```diff
@@ -148,23 +152,57 @@ class ReviewInline(admin.StackedInline):
     verbose_name_plural = "Reviews"
 
 
+def build_registration_data_from_application(application):
+    """
+    Builds registration_data dictionary from application so that it can be used in creating personalisation in a
+    standard way
+
+    :param application: The application object fetched from the data model
+
+    :return: A dictionary of registration data
+    """
+    registration_data = {
+        "registrar_name": application.registrar_person.name,
+        "domain_name": application.domain_name,
+        "registrant_type": application.registrant_org.type,
+        "domain_purpose": application.domain_purpose,
+        "exemption": "yes" if application.policy_exemption_evidence else "no",
+        "written_permission": "yes"
+        if application.written_permission_evidence
+        else "no",
+        "minister": "yes" if application.ministerial_request_evidence else "no",
+        "registrant_organisation": application.registrant_org.name,
+        "registrant_full_name": application.registrant_person.name,
+        "registrant_phone": str(application.registrant_person.phone_number),
+        "registrant_email": application.registrant_person.email_address,
+        "registrant_role": application.registry_published_person.role,
+        "registrant_contact_email": application.registry_published_person.email_address,
+    }
+    return registration_data
+
+
 def send_approval_or_rejection_email(request):
     """
     Sends Approval/Rejection mail depending on the action ( approval/rejection ) in the request object
 
     :param request: Request object
     """
     application = Application.objects.get(pk=request.POST["obj_id"])
-    registrar_name = application.registrar_person.name
     registrar_email = application.registrar_person.email_address
-    personalisation = {
-        "first_name": registrar_name,
-    }
+    reference = application.reference
+
+    # Build registration data dictionary to pass it to create_personalisation method
+    registration_data = build_registration_data_from_application(application)
```

Comment:
I would remove `build_` from the function name. It sounds like it has side effects, when it is a pure function. So for readability I'd call it just `registration_data_from_application`.
Same with `create_` below and `derive_`.

Reply:
Done